google/ggrc-core
# GGRC-2617,3947,3033 Update AutoStatusChangeable with new rules for making assessment 'In progress'

# Issue description

Add/update business rules for the automatic status update after performing some actions like Add/Update assessment field, Map/Unmap snapshots, etc. Details in the specification.

# Steps to test the changes
1. Create assessment with initial status.
2. Perform action from the specification.
3. Check that status was changed.
4. Run integration tests (test_autostatuschangable.py, test_autostatuschangable_custom_attributes.py)

# Solution description
The AutoStatusChangeable mixin was refactored to fit new requirements. Added the mapping to easy maintain rules in future. Added additional handlers. The code covered by the integration tests.
Removed unused/unnecessary code.

**I have one concern**: I've deleted [handle_custom_attribute_save](https://github.com/google/ggrc-core/commit/c3c00f4648d1eff45f848556b19518923294a57e) it seems unused. Please confirm that 'legacy API' not used anymore.

# Sanity checklist




# PR Review checklist
